app_dart/lib/src/service/branch_service.dart

```diff
@@ -101,9 +101,15 @@ class BranchService {
     final Iterable<GerritCommit> recipeCommits =
         await gerritService.commits(recipesSlug, Config.defaultBranch(recipesSlug));
     log.info('$recipesSlug commits: $recipeCommits');
-    final GithubService githubService = await config.createDefaultGitHubService();
     final List<gh.RepositoryCommit> githubCommits = await retryOptions.retry(
-      () async => await githubService.listCommits(Config.flutterSlug, branch, null),
+      () async {
+        final GithubService githubService = await config.createDefaultGitHubService();
```

Comment:
It would be good if this logic was contained in `GithubService`. Any call that returns a 403/401 should be retried with a newly minted client. Feel free to leave a TODO here on this, i'm not sure how needed this is in the code, and it would require a separate PR to implement